fortex/health/processors/coreference_processor.py

```diff
@@ -0,0 +1,209 @@
+            )
+
+        model = configs.model
+        cfg_inference = {
+            "greedyness": configs.greedyness,
+            "max_dist": configs.max_dist,
+            "max_dist_match": configs.max_dist_match,
+            "blacklist": configs.blacklist,
+            "store_scores": configs.store_scores,
+            "conv_dict": configs.conv_dict,
+        }
+        neuralcoref.add_to_pipe(
+            self.spacy_nlp, model=model, cfg_inference=cfg_inference
+        )
+
+    def initialize(self, resources: Resources, configs: Config):
+        super().initialize(resources, configs)
+        self.set_up(configs)
+
+    def _process(self, input_pack: DataPack):
+        r"""
+        Coreference resolution is done by
+        a spaCy pipeline with `NeuralCoref` added.
+
+        Then we translate the output to `CoreferenceGroup` and
+        `MedicalEntityMention`
+        """
+
+        def load_module(string):
+            path_str, module_str = string.rsplit(".", 1)
+            mod = importlib.import_module(path_str)
+            return getattr(mod, module_str)
+
+        # Default: Document
+        entry_type = load_module(self.configs.entry_type)
+
+        # Default: MedicalEntityMention
+        mention_type = load_module(self.configs.mention_type)
+
+        for entry_specified in input_pack.get(entry_type=entry_type):
+            result = self.spacy_nlp(entry_specified.text)
+
+            article = MedicalArticle(
+                pack=input_pack,
+                begin=entry_specified.span.begin,
+                end=entry_specified.span.end,
+            )
+
+            if not result._.has_coref:
+                article.has_coref = False
+                article.coref_groups = []
+                article.coref_resolved = result._.coref_resolved
+                article.coref_scores = {}
+            else:
+                article.has_coref = True
+                article.coref_groups = []
+                article.coref_resolved = result._.coref_resolved
+                article.coref_scores = result._.coref_scores
+                for cluster in result._.coref_clusters:
+
+                    mentions = []
+                    for mention in cluster.mentions:
+                        mention = mention_type(
+                            input_pack,
+                            mention.start_char,
+                            mention.end_char,
+                        )
+                        mentions.append(mention)
+
+                    group = CoreferenceGroup(input_pack)
+                    group.add_members(mentions)
+
+                    article.coref_groups.append(group)
+
+    @classmethod
+    def default_configs(cls):
+        r"""
+        This defines a basic config structure for `CoreferenceProcessor`.
+
+        Following are the keys for this dictionary:
+         - `entry_type`: Input entry type. Default: `"ft.onto.base_ontology.Document"`.
```

Comment:
A lot of configs here and the explanation is not particularly clear. For the Forte configs (entry_type, mention_type), please explain how they will be used. For the neuralcoref specific configs, try to group them together and provide a link to the official documentation

Reply:
Got it.

Reply:
Use case:
1. `entry_type`: as mentioned in the source code, you can do sentence level coref resolution by setting `entry_type` to `Sentence`. This usage is included in unit test (`test_inputs_and_entry_types`).
2. `mention_type`: if a user defines a subclass of `EntityMention` and adds some features, like adding multiple `ner_type`s to different data bases, he/she can just enjoy the feature by setting this parameter to his/her new subclass.

Reply:
neuralcoref specific configs are now in `get_default_cfg_inference`. `default_configs` now calls this method in a recursive manner.